src/pages/iou/steps/NewRequestAmountPage.tsx

```diff
@@ -14,66 +15,45 @@ import * as CurrencyUtils from '@libs/CurrencyUtils';
 import * as IOUUtils from '@libs/IOUUtils';
 import * as MoneyRequestUtils from '@libs/MoneyRequestUtils';
 import Navigation from '@libs/Navigation/Navigation';
-import {iouDefaultProps, iouPropTypes} from '@pages/iou/propTypes';
-import reportPropTypes from '@pages/reportPropTypes';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
 import * as IOU from '@userActions/IOU';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {IOU as IOUType, Report} from '@src/types/onyx';
 import MoneyRequestAmountForm from './MoneyRequestAmountForm';
 
-const propTypes = {
-    /** React Navigation route */
-    route: PropTypes.shape({
-        /** Params from the route */
-        params: PropTypes.shape({
-            /** The type of IOU report, i.e. bill, request, send */
-            iouType: PropTypes.string,
-
-            /** The report ID of the IOU */
-            reportID: PropTypes.string,
-
-            /** Selected currency from IOUCurrencySelection */
-            currency: PropTypes.string,
-        }),
-    }).isRequired,
-
-    /** The report on which the request is initiated on */
-    report: reportPropTypes,
-
-    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
-    iou: iouPropTypes,
-
-    /** The current tab we have navigated to in the request modal. String that corresponds to the request type. */
-    selectedTab: PropTypes.oneOf(_.values(CONST.TAB_REQUEST)),
+type NewRequestAmountPageOnyxProps = {
+    iou: OnyxEntry<IOUType>;
+    report: OnyxEntry<Report>;
+    selectedTab: OnyxEntry<string>;
 };
 
-const defaultProps = {
-    report: {},
-    iou: iouDefaultProps,
-    selectedTab: CONST.TAB_REQUEST.MANUAL,
-};
+type NewRequestAmountPageProps = NewRequestAmountPageOnyxProps & StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.AMOUNT>;
 
-function NewRequestAmountPage({route, iou, report, selectedTab}) {
+function NewRequestAmountPage({route, iou, report, selectedTab}: NewRequestAmountPageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
-    const prevMoneyRequestID = useRef(iou.id);
-    const textInput = useRef(null);
+    const prevMoneyRequestID = useRef(iou?.id);
+    const textInput = useRef<RNTextInput | null>(null);
 
-    const iouType = lodashGet(route, 'params.iouType', '');
-    const reportID = lodashGet(route, 'params.reportID', '');
+    const iouType = route.params.iouType ?? '';
+    const reportID = route.params.reportID ?? '';
     const isEditing = Navigation.getActiveRoute().includes('amount');
-    const currentCurrency = lodashGet(route, 'params.currency', '');
-    const isDistanceRequestTab = MoneyRequestUtils.isDistanceRequest(iouType, selectedTab);
+    const currentCurrency = route.params.currency ?? '';
```

Comment:
WHy are we defaulting this values to `''`? According to the navigation types they are always defined

Reply:
@fabioh8010 You're right! I've removed it!